SoftLayer/managers/load_balancer.py

```diff
@@ -92,15 +92,30 @@ def update_lb_health_monitors(self, uuid, checks):
     def get_lbaas_uuid_id(self, identifier):
         """Gets a LBaaS uuid, id. Since sometimes you need one or the other.
 
-        :param identifier: either the LB Id, or UUID, this function will return both.
+        :param identifier: either the LB Id, UUID or Name, this function will return UUI and LB Id.
         :return (uuid, id):
         """
-        # int objects don't have a len property.
-        if not isinstance(identifier, int) and len(identifier) == 36:
-            this_lb = self.lbaas.getLoadBalancer(identifier, mask="mask[id,uuid]")
+        mask = "mask[id,uuid]"
+        if isinstance(identifier, int):
+            this_lb = self.lbaas.getObject(id=identifier, mask=mask)
+        elif len(identifier) == 36 and utils.UUID_RE.match(identifier):
+            this_lb = self.lbaas.getLoadBalancer(identifier, mask=mask)
         else:
-            this_lb = self.lbaas.getObject(id=identifier, mask="mask[id,uuid]")
-        return this_lb['uuid'], this_lb['id']
+            this_lb = self.get_lbaas_by_name(identifier, mask=mask)
+
+        return this_lb.get('uuid'), this_lb.get('id')
+
+    def get_lbaas_by_name(self, name, mask=None):
+        """Gets a LBaaS by name.
+
+        :param name: Name of the LBaaS instance
+        :param mask:
+        :returns: SoftLayer_Network_LBaaS_LoadBalancer or an empty dictionary if the name is not found.
+        """
+        object_filter = {'name': {'operation': name}}
+        this_lbs = self.lbaas.getAllObjects(filter=object_filter, mask=mask)
+
+        return this_lbs[0] if this_lbs else {}
```

Comment:
Lets raise an exception here if we can't find a lbaas instance. This specific service returns a 500 error instead of a 404 which might confuse users.

Something like this I think would be ideal.
```
        if not this_lbs:
            raise exceptions.SoftLayerError("Unable to find LBaaS with name: {}".format(name))
        return this_lbs[0]
```